holoviews/plotting/bokeh/element.py

```diff
@@ -618,6 +638,138 @@ def framewise(self):
+                                             'orientation':'horizontal'}},
+                      'top_right':   {'pos': 'center',
+                                      'opts': {'location': 'top_right'}},
+                      'top_left':    {'pos': 'center',
+                                      'opts': {'location': 'top_left'}},
+                      'bottom_left': {'pos': 'center',
+                                      'opts': {'location': 'bottom_left',
+                                               'orientation': 'horizontal'}},
+                      'bottom_right': {'pos': 'center',
+                                      'opts': {'location': 'bottom_right',
+                                               'orientation': 'horizontal'}}}
+
+    colorbar = param.Boolean(default=False, doc="""
+        Whether to display a colorbar.""")
+
+    colorbar_position = param.ObjectSelector(objects=list(colorbar_specs),
+                                             default="right", doc="""
+        Allows selecting between a number of predefined colorbar position
+        options. The predefined options may be customized in the
+        colorbar_specs class attribute.""")
+
+    colorbar_opts = param.Dict(default={}, doc="""
+        Allows setting specific styling options for the colorbar overriding
+        the options defined in the colorbar_specs class attribute. Includes
+        location, orientation, height, width, scale_alpha, title, title_props,
+        margin, padding, background_fill_color and more.""")
+
+    logz  = param.Boolean(default=False, doc="""
+         Whether to apply log scaling to the z-axis.""")
+
+    _update_handles = ['color_mapper', 'source', 'glyph']
+
+    def _draw_colorbar(self, plot, color_mapper):
+        if LogColorMapper and isinstance(color_mapper, LogColorMapper):
+            ticker = LogTicker()
+        else:
+            ticker = BasicTicker()
+        cbar_opts = dict(self.colorbar_specs[self.colorbar_position])
+
+        # Check if there is a colorbar in the same position
+        pos = cbar_opts['pos']
+        if any(isinstance(model, ColorBar) for model in getattr(plot, pos, [])):
+            return
+
+        opts = dict(cbar_opts['opts'], bar_line_color='black',
+                    label_standoff=8, major_tick_line_color='black')
+        color_bar = ColorBar(color_mapper=color_mapper, ticker=ticker,
+                             **dict(opts, **self.colorbar_opts))
+
+        plot.add_layout(color_bar, pos)
+        self.handles['colorbar'] = color_bar
+
+
+    def _get_colormapper(self, dim, element, ranges, style):
+        low, high = ranges.get(dim.name)
+        palette = mplcmap_to_palette(style.pop('cmap', 'viridis'))
+        colormapper = LogColorMapper if self.logz else LinearColorMapper
+        cmapper = colormapper(palette, low=low, high=high)
+        if 'color_mapper' not in self.handles:
+            self.handles['color_mapper'] = cmapper
+        return cmapper
```

Comment:
Yes, there's some annoyances I have with this, ideally the handles would be set elsewhere but since it is now called from get_data, which is the main method the user has to implement and is duplicated everywhere I didn't want to move it out. The other thing is only the first colormapper that's created is actually used while the rest are simply used to update the existing colormapper. I'll leave it up to you if you think I should find a cleaner solution otherwise I'll make sure to leave a comment.